src/buildblock/Scanner.cxx

```diff
@@ -924,6 +925,29 @@ set_detector_map( const DetectorCoordinateMap::det_pos_to_coord_type& coord_map
       error("Scanner:set_detector_map: inconsistent number of detectors");
 }
 
+void
+Scanner::
+set_max_FOV_radius() {
+  if (!this->detector_map_sptr)
+  {
+    // for cylindrical scanners, all detectors have the same distance from the rotational axis
+    max_FOV_radius = inner_ring_radius;
+  }
+  else
+  {
+    // for other geometries, loop through all detectors and set the radius to the largest found distance
+    max_FOV_radius = inner_ring_radius;
+    for (auto tangential_pos = 0; tangential_pos < detector_map_sptr->get_num_tangential_coords(); tangential_pos++)
+      for (auto axial_pos = 0; axial_pos < detector_map_sptr->get_num_axial_coords(); axial_pos++)
+        for (auto radial_pos = 0; radial_pos < detector_map_sptr->get_num_radial_coords(); radial_pos++)
+    {
+      auto coord = detector_map_sptr->get_coordinate_for_det_pos(stir::DetectionPosition<>(tangential_pos, axial_pos, radial_pos));
+      const auto detector_radius = sqrt(coord.x() * coord.x() + coord.y() * coord.y()) - average_depth_of_interaction;
```

Comment:
hmmm. are we sure about this? is the detector-map giving "internal" points or surface points? This could be important for the blocks case, where we're suppose to pass the internal "radius".

Reply:
Yes, I was also confused by this, but it really is the internal position including DOI. Does this mean there might be a problem in the BlocksOnCylindrical implementation?

Reply:
https://github.com/UCL/STIR/blob/2945b72f9974a658f866fc2f47bb07ea869e0182/src/buildblock/GeometryBlocksOnCylindrical.cxx#L93
This is where the radius for the detector-map is set for BoC: it uses get_effective_ring_radius, not get_inner_ring_radius.

Reply:
ok. Let's document this then in `Scanner`: detector map stores "effective detection positions"  (i.e. approximately including DOI) for each crystal.

Reply:
added a comment in the scanner.h file where the detector map is defined.